eng/pipelines/branch-cleanup.yml

```diff
@@ -22,13 +28,37 @@ jobs:
       - ${{ each repo in parameters.Repos }}:
         - task: PowerShell@2
           displayName: Clean Up Sync Common Branches
-          condition: succeeded()
+          condition: succeededOrFailed()
+          continueOnError: true
+          inputs:
+            pwsh: true
+            workingDirectory: $(System.DefaultWorkingDirectory)
+            filePath: $(System.DefaultWorkingDirectory)/eng/common/scripts/Delete-RemoteBranches.ps1
+            arguments: >
+              -RepoId ${{ repo }}
+              -BranchPrefix "sync-eng/common-" 
+              -AuthToken $(azuresdk-github-pat)
+           
+      - pwsh: | 
+          $expirationDate = (Get-Date).AddDays(-7)
+          if ($env:LASTCOMMITDATEOVERRIDE) 
+          { 
+            $expirationDate = [datetime]$env:LASTCOMMITDATEOVERRIDE; 
+          } 
+          Write-Host "Using expiration date: $expirationDate" 
+          Write-Host "###vso[task.setvariable variable=ExpirationDate]$($expirationDate.ToString('o'))"
+        displayName: Set date into env variable
+      - ${{ each repo in parameters.DailyBranchRepos }}:
+        - task: PowerShell@2
+          displayName: Clean Up Docs Daily Branches
+          condition: succeededOrFailed()
           inputs:
             pwsh: true
             workingDirectory: $(System.DefaultWorkingDirectory)
             filePath: $(System.DefaultWorkingDirectory)/eng/common/scripts/Delete-RemoteBranches.ps1
             arguments: >
-              -RepoOwner "Azure"
-              -RepoName ${{ repo }}
-              -BranchPrefix "sync-eng/common-"
+              -RepoId ${{ repo }}
+              -BranchPrefix "daily"
+              -LastCommitOlderThan $(ExpirationDate) 
```

Comment:
Why cann't we just pass in the `(Get-Date).AddDays(-7)` here?

Reply:
The date here is changing according to the date. So every build might delete on a different date. 

The variable here is to control the date, so we do not need to have delete the new branches in testing phase.

Reply:
It is just for testing purpose.

Reply:
After a second thought, it is weird to add extra step for testing use. I removed the step.